element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java

```diff
@@ -29,12 +34,16 @@
 	/**
 	 * Creates a new endpoint context from DTLS session parameters.
 	 * 
+	 * @param peerAddress peer address of endpoint context
+	 * @param peerIdentity peer identity of endpoint context
 	 * @param sessionId the session's ID.
 	 * @param epoch the session's current read/write epoch.
 	 * @param cipher the cipher suite of the session's current read/write state.
-	 * @throws NullPointerException if any of the params is <code>null</code>.
+	 * @throws NullPointerException if any of the params (except peerIdentity) is {@code null}.
 	 */
-	public DtlsEndpointContext(String sessionId, String epoch, String cipher) {
+	public DtlsEndpointContext(InetSocketAddress peerAddress, Principal peerIdentity, String sessionId, String epoch,
```

Comment:
Should we add a toString to DTLSEndpointContext ? 
```java
@Override
public String toString() {
	return String.format("DTLS(%s:%d)", getPeerAddress().getHostString(), getPeerAddress().getPort());
}
```
Maybe with more detail ? (sessionId ?, cipher ?)

Reply:
This would be in danger of overload the logging.

Reply:
why not for overloading logging.

But why not adding `toString` to DTLSEndpointContext knowing that all other have a custom `toString` ? (I mean a too string with just peerAddress and Principal like other endpointContext)

Reply:
Uups, after changing to much, I assumed that it was already added and your comment was just about the extension. Sure, "toString()" is a that goog idea, that I thought it's already in :-).
I added also the session id but latter I would trunc this to the first characters.